libActivities

scunz opened the pull request:
Author @antis81:
To follow discussion and implementation status of all the related PR's, please head over to [MGV-Redmine](https://redmine.macgitver.org/issues/250).

scunz committed:
Move some RepoMan code around

scunz committed:
Make AutoRefresher internal

scunz committed:
Change RepoMan's "Set" to "List" everywhere

scunz committed:
Split off RepoMan & make everything compile at least

scunz committed:
Move refresher instance into MacGitver

scunz committed:
Repoman Events

scunz committed:
Fix some Compat-Event compile issues

scunz committed:
Logging Channel for Git

scunz committed:
Temporary code

scunz committed:
Core files for libActivities

scunz committed:
Snapshot of a mostly complete Activities interface

Still lacking lots of implementation and a good testcase.

antis81 commented:
@scunz I just cleaned up the issues, putting everything in the right places (well, technically :smile:).

How about setting up a "real" integration branch like `integrate/activities` based on this one?

In theory, we could use those branches to built and check preview releases (milestones) by Jenkins.

antis81 commented:
:zap: Build fails:

``` bash
/home/nils/Projects/MacGitver/Libs/libActivities/Activity.cpp:79:13: error: use of undeclared identifier 'assert'
            assert(running + success > 0);
            ^
```

@scunz Something's missing?

scunz commented:
You're on linux, right?

scunz committed:
Should fix Linux build

antis81 commented:
Yes, I'm on Linux. There's a few more compile errors of the same kind after fixing this one. Further some typos cause problems in the include section on case sensitive file systems. Shall I commit the fixes to the sacu/activities branch?

Btw.: Why not use Q_ASSERT macro instead?

scunz committed:
Add Service Runner into the backend

scunz committed:
Add: OpenRepository Service

scunz committed:
Add: RefreshRepository Service

scunz committed:
Move AutoRefresher out of RM-namespace

scunz committed:
Invoke OpenRepository service

scunz committed:
Add: refresh() method to RepoMan

scunz committed:
Move RepoMan Config Page to Config-Area

scunz committed:
Fixup: Remove unused code

scunz committed:
Step creation

scunz committed:
Fixup: Remove deadlock

scunz committed:
Actually implement state getters in Activity

scunz committed:
Fixup: "return {}" to avoid naming the constructor

scunz committed:
Public method to request opening of Progress Dialog

scunz committed:
Fixup: Inline comment (Manager Priv.hpp)

scunz committed:
Fixup to "Temporary Code" :(

scunz committed:
Turn RM::ObjType into an Enum Class

scunz committed:
Adjustments

scunz committed:
Move BasePrivate to own .cpp file

scunz commented:
Yes, please commit them on top of this (Not sure if you already did, but if so, rebase and push again, please).

I've just rebased everything once more and force-push'ed it (without seeing your comment). There's obviously something wrong with github and my mail...

antis81 committed:
Activities: use Q_ASSERT instead of assert (fix build on linux)

antis81 committed:
RepoMan: use Q_ASSERT instead of assert (fix build on linux)

antis81 committed:
RepoMan: fix include pathes for case sensitive file systems

scunz committed:
Be explicit about namespace of Dumper class

When we'll move most of Internal to Data, the Dumper class
resides where it is. So this'll make things easier later on

scunz committed:
Move Branch & Head Private's to own cpp

antis81 committed:
RepoMan: fix missing include

antis81 commented:
@scunz Hm ... the other fixes belong to libRepoMan. Should I commit them here anyways?

scunz commented:
Again, no notification per email. Think I've to ask GH Support for that one now.

@antis81, yes, commit them here to not-block yourself. But bear in mind that I'm in the process of un-fucking my mistakes in the repoman and I'm constantly rebasing it, so I can get the contents of the 3 "big" commits separated out. While I'm doing that, it's likely that there will be new temporary breakage introduced and some will vanish occasionally. The goal is to have every commit in repoman compile by itself.

So, if you commit repoman fixes here, I'll take care that repoman is changed in a way that these problems don't exist in the first place - but to do that earlier I need to be aware of them; otherwise they'll probably not be fixed before I'm at that stage...

scunz commented:
rebased again.

In addition to my previous comment: I.e. spot all the commits with "Fixup" in their name. They're meant to be moved into some other commit at some point.
